rust/arrow/src/buffer.rs

```diff
@@ -541,6 +552,18 @@ impl MutableBuffer {
             offset: 0,
         }
     }
+
+    /// View buffer as typed slice.
+    pub fn typed_data_mut<T: ArrowNativeType + num::Num>(&mut self) -> &mut [T] {
+        assert_eq!(self.len() % mem::size_of::<T>(), 0);
+        assert!(memory::is_ptr_aligned::<T>(self.raw_data() as *const T));
+        unsafe {
+            from_raw_parts_mut(
+                self.raw_data() as *mut T,
```

Comment:
use raw_data_mut()

Reply:
This fix has disappeared, I can't see it in the diff anymore 😕

Reply:
I opened [ARROW-8479](https://issues.apache.org/jira/browse/ARROW-8479)